### PR TITLE
[MERGE] sales_team, crm, crm_iap_*: improve reporting usability

### PR DESCRIPTION
**Pull request by ija-odoo**
PURPOSE

Purpose of this merge is to improve reporting usability
and views in crm, lead generation and iap services.

SPECIFICATIONS

Some usability improvements going to be done by this merge are listed
below.

Improve the action helpers for the actions bound to below listed menus
for better usability:

  * My Activities
  * Leads
  * Leads Analysis

Along with that, this commit also improves the following:

  * while generating leads, make 'Industries' field required on the modal so
    that results obtained are actionable instead of random leads;
  * after trying to generate leads, if no result is found, the same pop-up
    opens again with same criteria and displays a waring in the pop-up. It
    informs users that leads are not found and no credits are spent for their
    last search;
  * improve the warning message for unsuccessful enrichment due to invalid
    email;

This merge adds the color_picker widget for setting valid color of
CRM tags in the form view.

With this merge, the 'Enrich' button on lead now appears only after the
email is set and valid, which previously appeared even if no email is set
or if the value is not a valid email.

Right now, for the fetched contacts through enrichment, we display
name and title in separate columns (in the chatter), that leaves
lesser space for email and phone.

With this commit, we merge name and title columns, and provide a bit
extra space to email and phone so that chances for them being wrapped
in a new line are less. Also, if phone number is not available in any
contacts, the 'Phone' column is hidden, giving even more space for the
Name/Title and Email columns.

LINKS

COM PR odoo/odoo#64075
ENT PR odoo/enterprise#15561
Task ID-2393253

### CODE REVIEW

File: addons/crm/models/crm_lead.py [hunk 1]
```diff
@@ -875,6 +875,9 @@ def redirect_lead_opportunity_view(self):
 
     @api.model
     def get_empty_list_help(self, help):
+        if help:
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py:878]: As this is some html content, you should probably use is_html_empty

File: addons/crm/models/crm_lead.py [hunk 2]
```diff
@@ -875,6 +875,9 @@ def redirect_lead_opportunity_view(self):
...
+            return help
```
Comment (tde-banana-odoo) [inline addons/crm/models/crm_lead.py:879]: Could be great to add a docstring to this get_empty_list_help to explain its new behavior

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py [hunk 3]
```diff
@@ -242,22 +242,58 @@ def action_submit(self):
 
     def action_get_lead_action(self):
         self.ensure_one()
+        try_again_action_id = self.env.ref('crm_iap_lead.action_crm_iap_lead_mining_request_try_again').id
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:245]: Should probably handle action being deleted by user. No need to crash if it has been removed.

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py [hunk 4]
```diff
@@ -242,22 +242,58 @@ def action_submit(self):
...
+                'default_contact_filter_type':last_mining_request.contact_filter_type,
+                'default_preferred_role_id':last_mining_request.preferred_role_id.id,
+                'default_role_ids':last_mining_request.role_ids.ids,
+                'default_seniority_id':last_mining_request.seniority_id.id,
+            }
+        }
+
     def action_get_opportunity_action(self):
         self.ensure_one()
+        try_again_action_id = self.env.ref('crm_iap_lead.action_crm_iap_lead_mining_request_try_again').id
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:289]: Same remark here about action

File: addons/sales_team/views/crm_tag_views.xml [hunk 5]
```diff
@@ -19,7 +19,7 @@
                     </div>
                     <group>
                         <group>
-                            <field name="color" required="True"/>
+                            <field name="color" required="True" widget="color_picker"/>
```
Comment (tde-banana-odoo) [inline addons/sales_team/views/crm_tag_views.xml:22]: Should be done in a specific commit

File: addons/crm_iap_lead_enrich/models/crm_lead.py [hunk 6]
```diff
@@ -24,7 +24,7 @@ def _compute_show_enrich_button(self):
             self.show_enrich_button = False
             return
         for lead in self:
-            if not lead.active or not lead.email_from or lead.iap_enrich_done or lead.reveal_id or lead.probability == 100:
+            if not lead.active or lead.email_state == 'incorrect' or lead.iap_enrich_done or lead.reveal_id or lead.probability == 100:
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead_enrich/models/crm_lead.py:27]: Should be done in a specific commit as a fix

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py [hunk 7]
```diff
@@ -242,22 +242,58 @@ def action_submit(self):
...
         action = self.env["ir.actions.actions"]._for_xml_id("crm.crm_lead_opportunities")
         action['domain'] = [('id', 'in', self.lead_ids.ids), ('type', '=', 'opportunity')]
         action['help'] = _("""<p class="o_view_nocontent_empty_folder">
-            No opportunities found
+            No leads found
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:293]: Method is named action_get_opportunity_action and is called on opportunities. Why did you change this helper ?

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py [hunk 8]
```diff
@@ -242,22 +242,58 @@ def action_submit(self):
...
+    def _action_try_again(self):
+        last_mining_request = self.env['crm.iap.lead.mining.request'].search([('user_id', '=', self.env.uid)], order='create_date desc', limit=1)
+        return {
+            'name': _('Generate Leads'),
+            'res_model': 'crm.iap.lead.mining.request',
+            'views': [[False, 'form']],
+            'target': 'new',
+            'type': 'ir.actions.act_window',
+            'context': {
+                'is_modal': True, 'default_state':'draft',
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:266]: Why not doing a copy ?

**Review by tde-banana-odoo — CHANGES_REQUESTED** [review]
Validation

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py (hunk 7 above)
Comment (dja-odoo) [reply addons/crm_iap_lead/models/crm_iap_lead_mining_request.py]: Because of the spec : https://www.awesomescreenshot.com/image/5721716/930ea29139499201f65b7505297779bc

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py (hunk 8 above)
Comment (dja-odoo) [reply addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:276]: Because this is a regular model, and and if we cancel the modal, we should not create a record.

Comment (tde-banana-odoo) [reply addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:276]: Indeed... strange model. Ok, probably to cleanup in a near (?) future.

File: addons/crm_iap_lead/models/crm_iap_lead_mining_request.py (hunk 7 above)
Comment (tde-banana-odoo) [reply addons/crm_iap_lead/models/crm_iap_lead_mining_request.py:302]: Still you are showing opportunities and coming from opportunities. Strange, to check with PO.

File: addons/crm_iap_lead/views/crm_iap_lead_views.xml [hunk 9]
```diff
@@ -143,6 +143,14 @@
         <field name="view_mode">tree,form</field>
     </record>
 
+    <record id="action_crm_iap_lead_mining_request_try_again" model="ir.actions.server">
```
Comment (tde-banana-odoo) [inline addons/crm_iap_lead/views/crm_iap_lead_views.xml:146]: Guidelines !

**Review by tde-banana-odoo — COMMENTED** [review]
Hello,

Code looks globally ok, just one question left and a guidelines remark.

Thanks,

**Comment by tde-banana-odoo** [comment]
@robodoo r+ rebase-merge
